crucible-wasm/test/Test.hs

```diff
@@ -0,0 +1,153 @@
+import           Control.Lens ( (^.), (^?), _Right, to )
+import           Control.Monad ( unless, when )
+import           Data.Bifunctor ( first )
+import qualified Data.ByteString.Lazy as BSIO
+import qualified Data.ByteString.Lazy.Char8 as BSC
+import           Data.Char ( isLetter, isSpace )
+import           Data.List.Extra ( isInfixOf, isPrefixOf, stripPrefix )
+import           Data.Maybe ( catMaybes, fromMaybe, isJust )
+import qualified Data.Text as T
+import           Data.Versions ( Versioning, versioning, prettyV, major )
+import qualified GHC.IO.Exception as GE
+import           Numeric.Natural
+import           System.Environment ( withArgs, lookupEnv )
+import           System.Exit (ExitCode(..))
+import           System.FilePath ( (-<.>) )
+import qualified System.IO as IO
+
+import qualified Test.Tasty as TT
+import           Test.Tasty.HUnit ( testCase, assertFailure )
+import qualified Test.Tasty.Sugar as TS
+
+import qualified Crux
+
+import qualified Lang.Crucible.Wasm.Main as C
+
+
+cube :: TS.CUBE
+cube =
+  TS.mkCUBE
+  { TS.inputDirs = ["test-data/"]
+  , TS.rootName = "*.txt"
+  , TS.separators = "."
+  , TS.expectedSuffix = "good"
+  , TS.validParams =
+      [ ("solver", Just ["z3"])
+      ]
+  , TS.associatedNames =
+      [ ("config", "config")
+      , ("test-result", "result")
+      , ("skip", "skip")   -- when present, test config is skipped
+      ]
+  }
+
+main :: IO ()
+main = do
+  let cubes = [cube]
+  sweets <- concat <$> mapM TS.findSugar cubes
+  tests <- TS.withSugarGroups sweets TT.testGroup mkTest
+  let ingredients = TT.includingOptions TS.sugarOptions :
+                    TS.sugarIngredients cubes <>
+                    TT.defaultIngredients
+  TT.defaultMainWithIngredients ingredients $
+    TT.testGroup "crucible-wasm" tests
+
+-- Taken from crux-llvm test suite
+assertBSEq :: FilePath -> FilePath -> IO ()
+assertBSEq expectedFile actualFile = do
+  expected <- BSIO.readFile expectedFile
+  actual <- BSIO.readFile actualFile
+  let el = BSC.lines expected
+      al = BSC.lines actual
+  unless (el == al) $ do
+    let dl (e,a) = if e == a then db e else de e <> da a
+        db b = ["    F        |" <> b]
+        de e = ["    F-expect>|" <> e]
+        da a = ["    F-actual>|" <> a]
+    let details = concat $
+                  [ [ "MISMATCH for expected (" <> BSC.pack expectedFile <> ")"
+                    , "           and actual (" <> BSC.pack actualFile <> ") output:"
+                    ]
+                    -- Highly simplistic "diff" output assumes
+                    -- correlated lines: added or removed lines just
+                    -- cause everything to shown as different from
+                    -- that point forward.
+                  , concatMap dl $ zip el al
+                  , concatMap de $ drop (length al) el
+                  , concatMap da $ drop (length el) al
+                  ]
+    assertFailure $ BSC.unpack (BSC.unlines details)
```

Comment:
It might be worth factoring out some of this code into a `crux-test` library so that we don't have to duplicate this code. I think it's _just_ involved enough to where having to maintain it in two separate places would become annoying.

Reply:
I agree. Would you like to see that done in this PR, or in a follow-up?

Reply:
I'm fine with leaving that as follow-up work.

Reply:
https://github.com/GaloisInc/crucible/issues/1044